test/e2e/basic_provisioning_test.go

```diff
@@ -89,6 +94,22 @@ var _ = Describe("BMH Provisioning and Annotation Management", func() {
 		bmh.Spec.RootDeviceHints = &metal3api.RootDeviceHints{
 			DeviceName: "/dev/vda",
 		}
+		// The ssh check is not possible in all situations (e.g. fixture) so it can be skipped
+		if e2eConfig.GetVariable("SSH_CHECK_PROVISIONED") == "true" {
+			userDataSecretName := "user-data-secret"
+			metaDataSecretName := "meta-data-secret"
+			sshPubKeyPath := e2eConfig.GetVariable("SSH_PUB_KEY")
+			CreateUserdataSecret(ctx, clusterProxy.GetClient(), namespace.Name, userDataSecretName, sshPubKeyPath)
+			CreateMetadataSecret(ctx, clusterProxy.GetClient(), namespace.Name, metaDataSecretName)
```

Comment:
metadata has this:
```
metaDataContent := `{
		"instance-id": "cirros-bmo-e2e-test",
		"local-hostname": "cirros-bmo-test-vm"
	}`
```

I dont think we can skip it.

Reply:
Should we maybe have these as variables in the config file?

Reply:
Yeah I think we may need to add it there in some form. I'm a bit confused about this though. We definitely did not use any metadata before, so why would we not be able to skip it now?
Don't get me wrong, it is very good that we test metadata (we should check that it is really received in the BMH), but I don't think we must have any metadata.

Reply:
I know it is recieved by the BMH beause if you check the serial logs it will be asking for the password of `cirros-bmo-e2e-test` in the login screen instead of `cirros`. But I have not tested to just remove the metadata to se if it works. I can try that.

Reply:
You are right as usual :) It works fine when skipping meta-data.

Reply:
Nice to hear! Maybe skip it in this PR then? We could add it in a separate PR where we also add a check for it :slightly_smiling_face: